react-native-git-upgrade/checks.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const {execSync} = require('child_process');
+const semver = require('semver');
+
+function checkDeclaredVersion(declaredVersion) {
+  if (!declaredVersion) {
+    throw new Error(
+      'Your "package.json" file doesn\'t seem to have "react-native" as a dependency.'
```

Comment:
If you ship a plugin, these are already available inside react-native/local-cli as helpers and you should be able to just require them because of being inside node_modules.

Everything needed should be inside `local-cli/core`